demo/node/node_inprocess.go

```diff
@@ -50,40 +53,35 @@ type LocalNode struct {
 
 func NewLocalNode(i int, bindAddr string, cfg cfg.Config) *LocalNode {
 	nbase := path.Join(cfg.BasePath, fmt.Sprintf("node-%d", i))
-	os.MkdirAll(nbase, 0740)
+	_ = os.MkdirAll(nbase, 0740)
 	logPath := path.Join(nbase, "log")
 
-	// make certificates for the node.
-	err := httpscerts.Generate(
-		path.Join(nbase, fmt.Sprintf("server-%d.crt", i)),
-		path.Join(nbase, fmt.Sprintf("server-%d.key", i)),
-		bindAddr)
+	lg := log.New(nil, log.DebugLevel, false)
+
+	controlAddr := test.FreeBind(bindAddr)
+	dkgClient, err := net.NewDKGControlClient(lg, controlAddr)
 	if err != nil {
-		return nil
+		panic(err)
 	}
+
 	l := &LocalNode{
 		base:         nbase,
 		i:            i,
 		period:       cfg.Period,
-		tls:          cfg.WithTLS,
 		logPath:      logPath,
-		log:          log.NewLogger(nil, log.LogDebug),
+		log:          lg,
 		pubAddr:      test.FreeBind(bindAddr),
 		privAddr:     test.FreeBind(bindAddr),
-		ctrlAddr:     test.FreeBind("localhost"),
+		ctrlAddr:     controlAddr,
 		scheme:       cfg.Scheme,
 		beaconID:     cfg.BeaconID,
 		dbEngineType: cfg.DBEngineType,
 		pgDSN:        cfg.PgDSN,
 		memDBSize:    cfg.MemDBSize,
+		dkgRunner:    &test.DKGRunner{BeaconID: cfg.BeaconID, Client: dkgClient, Clock: clock.NewRealClock()},
 	}
 
-	var priv *key.Pair
-	if l.tls {
-		priv, err = key.NewTLSKeyPair(l.privAddr, nil)
-	} else {
-		priv, err = key.NewKeyPair(l.privAddr, nil)
-	}
+	priv, err := key.NewKeyPair(l.privAddr, l.scheme)
```

Comment:
Does this really works now? I'd expect it to fail because of the lack of proper TLS certificates, no, since now NewKeyPair is using TLS by default but we removed the certificate step above...